### DISCUSSION TITLE
new Canvas() is returning the html canvas

### DISCUSSION THREAD

**Discussion by HeyItsJs** [samizdatco/skia-canvas#66] [Q&A]
Just found out this awesome library while I was searching for an HTML to pdf converter library. Looks good so far. However, I am not able to get it running in React.js.

Here my code:
```js
import { Canvas } from "skia-canvas";

const canvas = new Canvas();
console.log("Canvas", canvas);
// This just prints the HTML canvas which doesn't have any methods like `saveAs`;
```

The same code if I run in a NodeJS project separately, it's working. But I am not able to get this working in a browser environment (React.js project). Any idea what am I missing?

**Comment by samizdatco** [comment]
If you're interested in using `skia-canvas`’s PDF-generation features, you'd have to run it server-side and use something like [express](https://expressjs.com) to handle the requests your react app makes. To generate PDFs on the client you'd be better off using something like [PDFKit](https://pdfkit.org) rather than this library which is primarily focused on emulating the browser’s `<canvas>` element for Node.js code.

The `Canvas` class that’s exposed to browsers is there purely to enable code that draws using `skia-canvas` on the server to also run in the browser.
